components/authn-service/server/grpc.go

```diff
@@ -14,19 +14,21 @@ import (
 
 	api "github.com/chef/automate/api/interservice/authn"
 	authz "github.com/chef/automate/api/interservice/authz/common"
+	authz_v2 "github.com/chef/automate/api/interservice/authz/v2"
 	tokens "github.com/chef/automate/components/authn-service/tokens/types"
 	"github.com/chef/automate/lib/grpc/health"
 	"github.com/chef/automate/lib/tracing"
 )
 
 type tokenAPI struct {
-	ts          tokens.Storage
-	authzClient authz.SubjectPurgeClient
+	ts                 tokens.Storage
+	authzSubjectClient authz.SubjectPurgeClient
+	authzV2Client      authz_v2.AuthorizationClient
 }
 
 // NewGRPCServer returns a server that provides our services: token
 // and authentication requests.
-func (s *Server) NewGRPCServer(authzClient authz.SubjectPurgeClient) *grpc.Server {
+func (s *Server) NewGRPCServer(authzSubjectClient authz.SubjectPurgeClient, authzV2Client authz_v2.AuthorizationClient) *grpc.Server {
```

Comment:
💭 [`authz_v2.AuthorizationClient`](https://github.com/chef/automate/blob/6c7875669ee458ccd58142a3e015f54baf7411b0/api/interservice/authz/v2/authz.pb.go#L596-L602) is quite a big interface. In the interest of making it simpler to mock this for tests, and of being more explicit about **what we need here**, we could introduce a smaller interface, 
```go
type ProjectChecker /* naming is hard */ interface {
    ValidateProjectAssignment(context.Context, *ValidateProjectAssignmentReq, ...grpc.CallOption) (*ValidateProjectAssignmentResp, error)
}
```
and use that instead. Any proper `authz_v2.AuthorizationClient` satisfies that, so we don't need to change a thing in the non-test code. Just a thought, though.